app/src/org/commcare/activities/LoginActivityUIController.java

```diff
@@ -347,16 +372,19 @@ protected LoginMode getLoginMode() {
         return loginMode;
     }
 
-    protected void setErrorMessageUI(String message) {
+    protected void setErrorMessageUI(String message, boolean showNotificationButton) {
         setLoginBoxesColorError();
 
         username.setCompoundDrawablesWithIntrinsicBounds(getResources().getDrawable(R.drawable.icon_user_attnneg), null, null, null);
         passwordOrPin.setCompoundDrawablesWithIntrinsicBounds(getResources().getDrawable(R.drawable.icon_lock_attnneg), null, null, null);
         loginButton.setBackgroundColor(getResources().getColor(R.color.cc_attention_negative_bg));
         loginButton.setTextColor(getResources().getColor(R.color.cc_attention_negative_text));
 
-        errorBox.setVisibility(View.VISIBLE);
-        errorBox.setText(message);
+        errorContainer.setVisibility(View.VISIBLE);
+        errorTextView.setText(message);
+        if(showNotificationButton) {
+            notificationButtonView.setVisibility(View.VISIBLE);
+        }
```

Comment:
`notificationButtonView.setVisibility(showNotificationButton  ? View.VISIBLE : View.GONE)` seems more safe to me in case `setErrorMessageUI` gets called twice with different values for `showNotificationButton`

Reply:
Agreed, that would also simplify the other logic. I'll make this change